### PR TITLE
Changed structure of parametrized functions 2D to allow more customization and flexibility

### PR DESCRIPTION
**Pull request by simonepeirone**
Changed structure of parametrized functions 2D following the procedure of parametrized functions 1D.

### CODE REVIEW

**Comment by simonepeirone** [comment]
I also modified line 43 in 04p2_linear_parametrizations.f90, removing

>  !! Notice that the derivatives above the first are not overridden since they are zero identically.

since this is not true anymore.

**Comment by EFTCAMB** [comment]
Hi Simone, great! Will check the PR asap!
There is some problem because the code is not compiling on Travis :)
The Travis build is failing on the branch because designer f(R) still does not pass legacy comparison but the problem here seems to be compilation.

**Comment by simonepeirone** [comment]
ok, the problem with compilation should be fixed now!

thanks :)

**Comment by EFTCAMB** [comment]
Perfect! Reviewing the code now!

File: eftcamb/05_abstract_parametrizations_2D.f90 [hunk 1]
```diff
@@ -226,21 +387,23 @@ subroutine ParametrizedFunction2DParameterNamesLatex( self, i, latexname )
...
+            latexname = self%param_names_latex(i)
+        else
+            latexname = TRIM(self%name)//'_'//integer_to_string(i-1)
+        end if
 
-    end subroutine ParametrizedFunction2DParameterValues
+    end subroutine ParametrizedFunction2DParameterNamesLatex
 
-    ! ---------------------------------------------------------------------------------------------
+    !-------------------------------------------------------------------------------------------------------
```
Comment (EFTCAMB) [inline eftcamb/05_abstract_parametrizations_2D.f90]: There seems to be a small formatting change here. Sorry for peakiness :)

**Comment by EFTCAMB** [comment]
Perfect! Just a minor (very minor) fix and PR is ready for merging.

**Comment by simonepeirone** [comment]
minor fix implemented!

**Pull request merged**